apps/st2-history/controller.js

```diff
@@ -253,6 +256,46 @@ angular.module('main')
       }
     };
 
+    $scope.rerun = (function () {
+      var rerun = $scope.$new();
+
+      rerun.metaSpec = {
+        type: 'object',
+        properties: {
+          ref: {
+            type: 'string',
+            name: 'Action'
+          }
+        }
+      };
+
+
+      rerun.open = function () {
+        $scope.$root.state.go('^.rerun', {id: $scope.record.id});
+        rerun.payload = _.clone($scope.payload);
```

Comment:
> I can't think of a good approach to handle that - I'm against using a special string and I also don't like the approach of including additional attribute in the payload.

There's two, actually, though they are pretty close.

We either need to add `merge=false` query param to rerun API preventing it from merging payload parameters with previous execution parameters and then use it in st2client.js. We've already added so much special cases, another one makes absolutely no difference.

Or we should stop design our API around st2client.py needs and make `merge=false` the default behavior, then merge parameters client side in case of st2client.py (where you would need to retype them manually and thus this is necessary) or just pre-fill them in inputs in case of st2web.

Traditionally, there is a third option of ignoring this rant until the need in this functionality arise =)